### PR TITLE
Refactor integration tests to allow execution in real vCenter

### PR DESCRIPTION
**Pull request by elsapassaro**
##### SUMMARY

- Adding default variables in integration targets in order to run integration tests on real vCenter instance. 
- Adding setup and teardown steps when needed
- Renaming integration targets that will run also in Jenkins CI  - adding the `vmware_` prefix

##### ISSUE TYPE

- Integration Tests Pull Request

##### ADDITIONAL INFORMATION

See [ACA-1511](https://issues.redhat.com/browse/ACA-1511)

### CODE REVIEW

**Comment by mariolenz** [comment]
OT but do you think there's a chance to use this "real vCenter instance" CI for [community.vmware](https://github.com/ansible-collections/community.vmware) and [vmware.vmware_rest](https://github.com/ansible-collections/vmware.vmware_rest), too?

Ref: ansible-collections/community.vmware#1746

**Review by bardielle — APPROVED** [review]
LGTM
@shellymiron please review it as well

**Comment by shellymiron** [comment]
> OT but do you think there's a chance to use this "real vCenter instance" CI for [community.vmware](https://github.com/ansible-collections/community.vmware) and [vmware.vmware_rest](https://github.com/ansible-collections/vmware.vmware_rest), too?
> 
> Ref: [ansible-collections/community.vmware#1746](https://github.com/ansible-collections/community.vmware/issues/1746)

Hey Mario :) we will have a discussion about it tomorrow and we will keep this PR updated with what was decided

**Comment by shellymiron** [comment]
@mariolenz hey!
we had a discussion about integrating the real vcenter CI within the community repo, and our thoughts is that it will be hard to control which PRs will be destructive for that env, since it serves multiple teams in redhat. 
regarding the vmware_rest repo, we will want to have it running on the real env, but this is a low priority for now, since it has a lot more features and we are only 2 QEs :) 
With that thoughts in mind, we would like to hear your opinion and if it is necessary/critical to have such job in the community

**Comment by mariolenz** [comment]
> With that thoughts in mind, we would like to hear your opinion and if it is necessary/critical to have such job in the community

@shellymiron Probably not. But integration tests for community.vmware are still run with vCenter and ESXi 7.0 U3. It's really about time to start running those with a more current version. Latest release is 8.0 U3 and I think 7.0 will be EOL next(?) year.

That said, I'm not unhappy with Zuul. It mostly works. But I understood that you don't want to work on those integration tests because of lacking the time / knowledge / experience / whatever to do this.

I think it's both critical and necessary to run integration tests with a more current vSphere version, preferably 8.0 U3. And instead of trying to find someone who can update the Zuul integration tests accordingly or help me with yet another solution, I thought it would be the best thing if I could just re-use your solution.

**Comment by shellymiron** [comment]
LGTM
looks good :)

File: tests/integration/targets/vmware_appliance_info/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+vcenter_port: 443
```
Comment (machacekondra) [inline tests/integration/targets/vmware_appliance_info/defaults/main.yml:1]: vcenter_port: 1080

Comment (elsapassaro) [reply tests/integration/targets/vmware_appliance_info/defaults/main.yml:1]: the variables in defaults/main.yml should be used for eco vcenter and they are overwritten by variable in [vars.yml](https://github.com/ansible-collections/vmware.vmware/pull/29/files#diff-575f9dc8e37da616efa732c1272fe792777998e5eea595078cff3ed17e3f65d0R6) when the role is imported from run.yml playbook. 

It looks like in the execution the port it's using is indeed 1080 but it fails:

```
TASK [vmware_appliance_info : Gather appliance info] ***************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failed to connect to vCenter or ESXi API at 127.0.0.1:1080 : 404 Client Error: Not Found for url: https://127.0.0.1:1080/api"}
```

**Review by machacekondra — APPROVED** [review]

**Pull request merged**